pyfstat/grid_based_searches.py

```diff
@@ -359,12 +366,12 @@ def _get_tolerance_from_savetxt_fmt(self):
                 )
         return rtol, atol
 
-    def save_array_to_disk(self, data):
+    def save_array_to_disk(self):
         logging.info("Saving data to {}".format(self.out_file))
         header = "\n".join(self.output_file_header)
-        header += "\n" + " ".join(self.keys)
-        outfmt = self.get_savetxt_fmt()
-        Ncols = np.shape(data)[1]
+        header += "\n" + " ".join(self.output_keys)
+        outfmt = list(self.get_savetxt_fmt().values())
```

Comment:
There was an "actually not Ok" regarding a similar line. I'm noticing quite a volume of duplicated lines (that's what we have, I guess...)

Reply:
for reference, that was this thread: https://github.com/PyFstat/PyFstat/pull/171#discussion_r516936239

Reply:
I think it didn't cause any errors for the grid-based searches, but to be on the safe side I've replicated the same fix as for MCMC here too: d6a50ffb771107809400bb71ac6826f402c96851